### PR TITLE
search by title e2e

### PR DESCRIPTION
**Pull request by js-kyle**
This is the e2e working but just need guidance on what we want to store & return for this

### CODE REVIEW

**Comment by SubJunk** [comment]
@js-kyle let's store and return the same as `getByOsdbHash`, probably using a shared function.

The way it's done at the moment is there is this feeder function https://github.com/UniversalMediaServer/UniversalMediaServer/blob/master/src/main/java/net/pms/util/OpenSubtitle.java#L1786 that passes different combinations of parameters into https://github.com/UniversalMediaServer/UniversalMediaServer/blob/master/src/main/java/net/pms/util/OpenSubtitle.java#L1822

Not saying that's how it should be done here but it's a reference.

**Comment by js-kyle** [comment]
@SubJunk ready to be reviewed

File: src/controllers/media.ts [hunk 1]
```diff
@@ -64,3 +74,40 @@ export const getByOsdbHash = asyncHandler(async(req: Request, res: Response) =>
   dbMeta = await MediaMetadata.create(newMetadata);
   return res.json(dbMeta);
 });
+
+export const getBySanitizedTitle = asyncHandler(async(req: Request, res: Response, next: NextFunction) => {
+  const { title, language = 'eng' } = req.body;
+
+  if (!title) {
+    return next(new Error('title is required'));
```
Comment (SubJunk) [inline src/controllers/media.ts]: If you change this line to `throw new Error('title is required');` does it still go through the `next` function or do we need to explicitly do this?

File: src/controllers/media.ts [hunk 2]
```diff
@@ -10,6 +10,21 @@ const MESSAGES = {
   openSubsOffline: 'OpenSubtitles API seems offline, please try again later',
 };
 
+export const FAILED_LOOKUP_SKIP_DAYS = 30;
+
+export const skipFailedLookup = async(dbQuery): Promise<boolean> => {
+  const recordFromFailedLookupsCollection: FailedLookupsInterface = await FailedLookups.findOne({ dbQuery });
+  if (!recordFromFailedLookupsCollection) {
+    return Promise.resolve(false);
```
[the remark this answers is not in the document]
Comment (js-kyle) [reply src/controllers/media.ts]: Ah yep that is correct, cleaned up now

File: src/controllers/media.ts [hunk 3]
```diff
@@ -10,6 +10,21 @@ const MESSAGES = {
   openSubsOffline: 'OpenSubtitles API seems offline, please try again later',
 };
 
+export const FAILED_LOOKUP_SKIP_DAYS = 30;
+
+export const skipFailedLookup = async(dbQuery): Promise<boolean> => {
```
[the remark this answers is not in the document]
Comment (js-kyle) [reply src/controllers/media.ts]: Renamed

File: src/controllers/media.ts (hunk 1 above)
Comment (js-kyle) [reply src/controllers/media.ts]: `thrown` errors are passed to the Express error handler, so same same, but this is probably cleaner so updated

**Comment by SubJunk** [comment]
Very nice

**Pull request merged**